apps/invoices/src/modules/app-configuration/app-configuration-router.ts

```diff
@@ -0,0 +1,115 @@
+import { createLogger } from "@saleor/apps-shared";
+import { z } from "zod";
+import { protectedClientProcedure } from "../trpc/protected-client-procedure";
+import { router } from "../trpc/trpc-server";
+import { createSettingsManager } from "./metadata-manager";
+import { AppConfigV2MetadataManager } from "./schema-v2/app-config-v2-metadata-manager";
+import { GetAppConfigurationV2Service } from "./schema-v2/get-app-configuration.v2.service";
+import { PrivateMetadataAppConfiguratorV1 } from "./schema-v1/app-configurator";
+import { AppConfigV2 } from "./schema-v2/app-config";
+import { ConfigV1ToV2Transformer } from "./schema-v2/config-v1-to-v2-transformer";
+import { AppConfigV1 } from "./schema-v1/app-config-v1";
+import { Client } from "urql";
+import { ConfigV1ToV2MigrationService } from "./schema-v2/config-v1-to-v2-migration.service";
+import { trpcClient } from "../trpc/trpc-client";
+
+// todo unify
+const upsertAddressSchema = z.object({
+  address: z.object({
+    city: z.string().min(1),
+    cityArea: z.string(),
+    companyName: z.string().min(1),
+    country: z.string().min(1),
+    streetAddress1: z.string().min(1),
+    streetAddress2: z.string(),
+    countryArea: z.string(),
+    postalCode: z.string().min(1),
+  }),
+  channelSlug: z.string(),
+});
+
+export const appConfigurationRouter = router({
+  fetchChannelsOverrides: protectedClientProcedure.query(async ({ ctx, input }) => {
+    const logger = createLogger({ saleorApiUrl: ctx.saleorApiUrl });
+
+    logger.debug("appConfigurationRouterV2.fetch called");
+
+    const appConfigV2 = await new GetAppConfigurationV2Service(ctx).getConfiguration();
+
+    /**
+     * MIGRATION CODE START - remove when metadata migrated
```

Comment:
Its tempting, but when I try to implement it, it gets complicated. 

With my original approach we have clear separation
1. Service (that will stay) to fetch V2 config
2. Service that migrates
3. Caller of the services that can modify data before saving it

If I merge it to single service, I see problems:
1. This will be removed anyway
2. It will hide mutating data behind "get" method :/ 
3. It must inject mutation data to get method OR mutate twice (migrate + modify later). If app breaks, we lose transaction

I think its not worth it

Reply:
Good points, lets leave it